bloom-housing/bloom
# feat(preferences): add delete preference button

# Pull Request Template

## Issue Overview #3028 

This PR addresses #issue


## Description

This PR adds the delete functionality to the "settings" page in the partners site. 

It adds the delete icon back in for each preference
![image](https://user-images.githubusercontent.com/42942267/190726530-7679de42-bf16-4b76-b355-94c41f66ea70.png)

Once clicked a modal pops up
![image](https://user-images.githubusercontent.com/42942267/190726319-e623ef47-bd07-4733-aa26-ba7126d4cc41.png)

And on Deletion a toast notification indicates the preference was deleted
![image](https://user-images.githubusercontent.com/42942267/190727303-88b27d1d-d6ac-429d-9549-df0922c04fa7.png)


## How Can This Be Tested/Reviewed?

There are two main use cases that should be tested. First off is the happy path of deleting a preference that doesn't have any listings attached. The second is testing on a preference that does have at least one listing attached to it. In that case the "Delete" button on the modal shouldn't be visible and there should be a different text in the modal.

You will need to run the migration script in order for this feature to work.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
